src/commands/sokoban.ts

```diff
@@ -0,0 +1,95 @@
+import { PathSrc } from '#lib/common/constants';
+import { LanguageKeys } from '#lib/i18n/LanguageKeys';
+import { buildGameControls, buildSokobanGameFromResolvableLevel, encodeResolvableLevel } from '#lib/utilities/sokoban';
+import { Command, RegisterCommand, RegisterSubcommand, type AutocompleteInteractionArguments } from '@skyra/http-framework';
+import { applyLocalizedBuilder, getSupportedLanguageT } from '@skyra/http-framework-i18n';
+import { MessageFlags } from 'discord-api-types/v10';
+import { readFile } from 'fs/promises';
+import { URL } from 'url';
+
+const Root = LanguageKeys.Commands.Sokoban;
+const Levels: Level[] = JSON.parse(await readFile(new URL('lib/common/levels.json', PathSrc), { encoding: 'utf-8' }));
+
+@RegisterCommand((builder) => applyLocalizedBuilder(builder, Root.RootName, Root.RootDescription))
+export class UserCommand extends Command {
+	@RegisterSubcommand((builder) =>
+		applyLocalizedBuilder(builder, Root.OptionPlayLevel) //
+			.addStringOption((builder) => applyLocalizedBuilder(builder, Root.OptionChooseLevel).setAutocomplete(true).setRequired(true))
+	)
+	public async level(interaction: Command.ChatInputInteraction, options: LoadLevelOptions) {
+		const t = getSupportedLanguageT(interaction);
+		const resolvableLevel = Levels.find((level) => level.name === (options.level ?? 'default'));
+		if (!resolvableLevel) {
+			return interaction.reply({
+				content: t(LanguageKeys.Commands.Sokoban.SokobanInvalidLevel),
+				flags: MessageFlags.Ephemeral
+			});
+		}
+		const levelResult = buildSokobanGameFromResolvableLevel(resolvableLevel.data);
+
+		if (levelResult.isErr()) {
+			return interaction.reply({
+				content: t(LanguageKeys.Commands.Sokoban.SokobanInvalidComponent, { value: levelResult.unwrapErr() }),
+				flags: MessageFlags.Ephemeral
+			});
+		}
+
+		const level = levelResult.unwrap();
+
+		return interaction.reply({
+			content: level.toString(),
+			components: buildGameControls(encodeResolvableLevel(level.components), level.checkPossibleMoves()),
+			flags: MessageFlags.Ephemeral
+		});
```

Comment:
This code seems to be duplicated with `custom-level`'s, please move this into a new function using `result.match` like:
```ts
const data = result.match({
	ok: (level) => ({
		content: level.toString(),
		components: buildGameControls(encodeResolvableLevel(level.components), level.checkPossibleMoves()),
		flags: MessageFlags.Ephemeral
	}),
	err: (error) => ({
		content: t(LanguageKeys.Commands.Sokoban.SokobanInvalidComponent, { value: error }),
		flags: MessageFlags.Ephemeral
	})
});
return interaction.reply(data);
```

That way you'll be able to avoid the `unwrap()` calls, which is something to be avoid where an alternative exists.